cep-??.md

```diff
@@ -0,0 +1,246 @@
+>  < 0.5
+>  < 0.9.6
+>  < 0.960923
+>  < 1.0
+>  < 1.1dev1    # special case 'dev'
+>  < 1.1_       # appended underscore is special case for openssl-like versions
+>  < 1.1a1
+>  < 1.1.0dev1  # special case 'dev'
+> == 1.1.dev1   # 0 is inserted before string
+>  < 1.1.a1
+>  < 1.1.0rc1
+>  < 1.1.0
+> == 1.1
+>  < 1.1.0post1 # special case 'post'
+> == 1.1.post1  # 0 is inserted before string
+>  < 1.1post1   # special case 'post'
+>  < 1996.07.12
+>  < 1!0.4.1    # epoch increased
+>  < 1!3.1.1.6
+>  < 2!0.4.1    # epoch increased again
+> ```
+>
+> Some packages (most notably openssl) have incompatible version conventions.
+> In particular, openssl interprets letters as version counters rather than
+> pre-release identifiers. For openssl, the relation:
+>
+> ```
+> 1.0.1 < 1.0.1a  =>  False  # should be true for openssl
+> ```
+>
+> holds, whereas conda packages use the opposite ordering. You can work-around
+> this problem by appending an underscore to plain version numbers:
+>
+> ```
+> 1.0.1_ < 1.0.1a =>  True   # ensure correct ordering for openssl
+> ```
+
+With that ordering in mind, the following operators are allowed:
+
+- Range operators: `<`, `>`, `<=`, `>=`. Note that `<1.0` would include `1.0a` given the ordering above!
+- Exact equality and negated equality: `==`, `!=`.
+- Fuzzy equality: `=`, `*`. `=1.0` and `1.0.*` are equivalent, and both would match `1.0.0` and `1.0.1`, but not `1.1` or `0.9`.
+- Logical operators: `|` means OR, `,` means AND. `1.0|1.2` would match both `1.0` and `1.2`. `>=1.0,<2.0a0` would match everything between `1.0` and the last version before `2.0a0`. `,` (AND) has higher precedence than `|` (OR). `>=1,<2|>3` means `(>=1,<2)|(>3)`; i.e. greater than or equal to `1` AND less than `2` or greater than `3`, which matches `1`, `1.3` and `3.0`, but not `2.2`.
+- Semver-like operator: `~=`. `~=0.5.3` is equivalent to `>=0.5.3, <0.6.0a` and this syntax is preferred for backwards compatibility.
+
+No spaces are allowed between operators. `1.8*` and `1.8.*` are equivalent, but the latter is preferred for clarity.
+
+### Exact matches
+
+To fully-specify a package record with a full, exact spec, these fields must be given as exact values: `channel` (preferrably by URL), `subdir`, `name`, `version`, `build`. Alternatively, an exact spec can also be given by `*[md5=12345678901234567890123456789012]` or `*[sha256=f453db4ffe2271ec492a2913af4e61d4a6c118201f07de757df0eff769b65d2e]`.
+
+## Examples
+
+```python
+>>> str(MatchSpec('foo 1.0 py27_0'))
+'foo==1.0=py27_0'
+>>> str(MatchSpec('foo=1.0=py27_0'))
+'foo==1.0=py27_0'
+>>> str(MatchSpec('conda-forge::foo[version=1.0.*]'))
+'conda-forge::foo=1.0'
+>>> str(MatchSpec('conda-forge/linux-64::foo>=1.0'))
+"conda-forge/linux-64::foo[version='>=1.0']"
+>>> str(MatchSpec('*/linux-64::foo>=1.0'))
+"foo[subdir=linux-64,version='>=1.0']"
+```
+
+## Reference
+
+- [`conda.models.match_spec.MatchSpec`](https://github.com/conda/conda/blob/24.5.0/conda/models/match_spec.py)
```

Comment:
Just FYI this is the rattler implementation: https://github.com/mamba-org/rattler/blob/main/crates/rattler_conda_types/src/match_spec/mod.rs